app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -208,10 +224,48 @@ class PipelineSampleReport extends React.Component {
       search_taxon_id: searchTaxonId,
       thresholded_taxons,
       selected_taxons,
+      selected_taxons_top: selected_taxons.slice(0,  this.max_rows_to_render),
+      pagesRendered: 1,
+      rendering: true,
       rows_passing_filters: selected_taxons.length
+    },
+    () => {
+      this.setState({rendering: false})
+    }
+    );
+  }
+
+  //Load more samples on scroll
+  scrollDown() {
+    var that = this;
+    $(window).scroll(function() {
+      if ($(window).scrollTop() > $(document).height() - $(window).height() - 6000) {
+        {that.state.rows_total > 0 && !that.state.rendering ? that.renderMore() : null }
+        return false;
+      }
     });
   }
 
+
+  renderMore() {
+    let selected_taxons = this.state.selected_taxons
+    let currentPage = this.state.pagesRendered
+    let rowsPerPage = this.max_rows_to_render
+    if (selected_taxons.length > currentPage * this.max_rows_to_render) {
+      let next_page = selected_taxons.slice(currentPage * rowsPerPage, rowsPerPage * (currentPage +1))
+      this.setState((prevState) => ({
+        selected_taxons_top: [...prevState.selected_taxons_top, ...next_page],
```

Comment:
so theoretically as you scroll down each render will take longer and longer?

Reply:
yep.